lib/src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -814,4 +814,10 @@ def b = new CpsTransformerTest.Base()
 return (b.&toString)() + (String.getClass().&getSimpleName)()
 ''') == "baseClass"
     }
+
+    @Issue("JENKINS-32213")
+    @Test
+    void allClassesSerializable() {
+        evalCPSonly('class C {}; def c = new C(); assert c instanceof Serializable')
```

Comment:
Fine but you also need to test classes which explicitly or implicitly implement `Serializable` already.

Also check behavior of local classes defined inside `@NonCPS` methods, if Groovy even “does” that. These should not be touched.